### ISSUE TITLE
Question about "h_filtered" and problem setting

### ISSUE THREAD

**Issue by kpmokpmo** [fmorenopino/SAAM#1]
Hi, good job!

Since my hardware only support latest version of pytorch, I am waiting for your updated code for torch1.8+ and didn't have a chance to run the current code. But It seems to me that only mu, sigma or h can affect the loss function in some way and the result term of sa module "h_filtered" has nothing to do with the training process in the Transformer-based model. So how it works?

And another question is about the problem setting. It seems you randomly pick some subsets from different original long series. It ensentially means we treat the data as some univariate series who have exactly same inherit pattern instead of a multivariate problem. This may or may not be ture even if those data are from sensors recording the same period of time. Would that make more sense if we apply the model to a single supper long series since the inherit pattern should be consisted?

Thanks very much

**Comment by fmorenopino** [comment]
Thanks for your feedback!

Actually, 'h_filtered' is just the filtered version of the original embedding. This filtered representation of the embedding is later used by the probabilistic model which performs the forecast (instead of the original embedding), thus making the parameters of the Spectral Attention module derivable. The log-likelihood optimizes then three different set of parameters: embedding, SA module (which produces "h_filtered") and probabilistic model.

Regarding your second question, the proposed model deals with univeriate data. The model characterizes the time series (also its covariates) in the spectral domain by considering each dimension of the embedding as independent realizations of the same process. This implies that, for the global spectral characterization, we are calculating a Monte Carlo approximation of the process's PSD. The idea behind this characterization is to summarize all global patterns visible in the data. The global attention model, which operates over the spectral domain, will later decide which of these patterns must be filtered out and which must be incorporated into the predictions. Working with a unique long series that repeatly shows certains patterns would ease spectral attention module's job.

I hope this clarifies your doubts. Regards.

PD: for extra information regarding how the model works, you can visit the arXiv pre-print: https://arxiv.org/abs/2107.05984.
